app/assets/javascripts/Components/__tests__/tag_modal.test.jsx

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import {render, screen, fireEvent, waitFor, cleanup} from "@testing-library/react";
+import TagModal from "../Helpers/tag_modal";
+import Modal from "react-modal";
+
+describe("TagModal", () => {
+  let props;
+  let component;
+
+  beforeEach(() => {
+    props = {
+      name: "",
+      description: "",
+      isOpen: true,
+      onRequestClose: jest.fn().mockImplementation(() => (props.isOpen = false)),
+      tagModalHeading: "Heading",
+      onSubmit: jest.fn().mockImplementation(() => (props.isOpen = false)),
+      handleNameChange: jest.fn().mockImplementation(() => {}),
+      handleDescriptionChange: jest.fn().mockImplementation(() => {}),
+    };
+
+    // Set the app element for React Modal
+    Modal.setAppElement("body");
+    component = render(<TagModal {...props} />);
+  });
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  it("should close on Cancel", () => {
+    fireEvent.change(screen.getByTestId("tag_name_input"), {target: {value: "Name"}});
+    fireEvent.click(screen.getByText(I18n.t("cancel")));
+    expect(props.onRequestClose).toHaveBeenCalledTimes(1);
+  });
+
+  it("should not be submittable with empty name", () => {
+    expect(screen.getByText(I18n.t("save"))).toBeDisabled();
+  });
+
+  it("should render correct headings, labels, buttons and inputs", () => {
+    const headingLabelDataTestIds = [
+      "tag_modal_heading",
+      "tag_name_label",
+      "tag_description_label",
+    ];
+    headingLabelDataTestIds.forEach(dataTestId => {
+      expect(screen.queryByTestId(dataTestId)).toBeInTheDocument();
+    });
+    const buttonDataTestIds = screen
+      .queryAllByRole("button", {hidden: true})
+      .map(button => button.getAttribute("data-testid"));
+    expect(buttonDataTestIds.sort()).toEqual(["tag_submit_button", "tag_cancel_button"].sort());
+    const inputDataTestIds = screen
+      .queryAllByRole("textbox", {hidden: true})
+      .map(textarea => textarea.getAttribute("data-testid"));
+    expect(inputDataTestIds.sort()).toEqual(["tag_name_input", "tag_description_input"].sort());
+    // verify maxLength for the inputs
+    expect(screen.getByTestId("tag_name_input")).toHaveAttribute("maxLength", "30");
+    expect(screen.getByTestId("tag_description_input")).toHaveAttribute("maxLength", "120");
+  });
+
+  describe("When name filled", () => {
+    beforeEach(() => {
```

Comment:
I'm not really sure what the best practice is for test organization is in this case, when you want to customize some props to overwrite what the default `beforeEach` is doing. The approach you're currently taking seems a bit clunky. There's a `rerender` method for react testing library, how about using that instead?

Reply:
Changed to using `rerender`, still some setup needed but does get rid of the extra `cleanup`